cra-network-aware-force-cache-loading/src/components/ConnectionAwareMedia/index.js

```diff
@@ -48,15 +47,47 @@ const ConnectionAwareMedia = () => {
 
     console.log('[ConnectionAwareMedia] effectiveConnectionType => ', effectiveConnectionType);
 
-    fetch(mediaURL, {cache: 'force-cache'})
-      .then(response => response.blob())
-      .then(blob => {
-          const reader = new FileReader();
-          reader.onload = function() {
-            setBase64Image(this.result); // `this.result` contains a base64 data URI
-          };
-          reader.readAsDataURL(blob);
-      });
+    const requestImage = async baseURL => {
+      const qualities = ['max-res', 'medium-res', 'min-res'];
+
+      const feedImage = blob => {
+        const reader = new FileReader();
+        reader.onload = function() {
+          setBase64Image(this.result); // `this.result` contains a base64 data URI
+        };
+        reader.readAsDataURL(blob);
+      };
+
+      let imageBlob;
+      for (const quality of qualities) {
+        const url = baseURL.replace(/max-res|medium-res|min-res/, quality);
+        try {
+          imageBlob = await fetch(url, {
+            cache: 'only-if-cached',
+            mode: 'same-origin'
+          }).then(response => {
+            return response.blob();
+          });
+          console.log('only-if-cached feeding url => ', url);
+          if (imageBlob) break;
```

Comment:
Based on [the docs](https://developer.mozilla.org/en-US/docs/Web/API/Request/cache#Value) for `only-if-cached`, I believe that a cache miss will result in a valid `Response` object, but one with a `status` of `504`. So I think this will end up `break`ing even if there is a cache miss.

That's why I suggested using [`response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) (which will be `false` if the `status` is `504`, but `true` if the `status` is `200`.

Reply:
Thank you for your feedback. @jeffposnick.

According to what I've tested, if a cache miss happens, the flow gets into catch block (`try catch`).
I think using response.ok makes more sense. I'll update the logic that way.